integration-tests/contract-xcm/lib.rs

```diff
@@ -0,0 +1,300 @@
+            let transfer_message = call_builder.transfer_through_xcm(
+                default_accounts::<DefaultEnvironment>().bob,
+                CONTRACT_BALANCE + 1,
+            );
+
+            let call_res = client
+                .call(&ink_e2e::alice(), &transfer_message)
+                .dry_run()
+                .await?
+                .return_value();
+
+            assert!(matches!(call_res, Err(RuntimeError::XcmExecuteFailed)));
+            Ok(())
+        }
+
+        #[ink_e2e::test(backend(runtime_only(sandbox = MockNetworkSandbox)))]
+        async fn xcm_send_works<Client: E2EBackend>(mut client: Client) -> E2EResult<()> {
+            use frame_support::traits::{
+                fungibles::Mutate,
+                tokens::currency::Currency,
+            };
+            use mock_network::{
+                parachain,
+                parachain_account_sovereign_account_id,
+                relay_chain,
+                ParaA,
+                Relay,
+                TestExt,
+                INITIAL_BALANCE,
+            };
+            use pallet_balances::{
+                BalanceLock,
+                Reasons,
+            };
+
+            let mut constructor = ContractXcmRef::new();
+            let contract = client
+                .instantiate("contract_xcm", &ink_e2e::alice(), &mut constructor)
+                .value(CONTRACT_BALANCE)
+                .submit()
+                .await
+                .expect("instantiate failed");
+
+            let account_id: &[u8; 32] = contract.account_id.as_ref();
+            let account_id: [u8; 32] = account_id.clone();
+            let account_id = account_id.into();
+
+            ParaA::execute_with(|| {
+                parachain::Balances::make_free_balance_be(&account_id, INITIAL_BALANCE);
+                parachain::Assets::mint_into(0u32.into(), &account_id, INITIAL_BALANCE)
+                    .unwrap();
+            });
+
+            Relay::execute_with(|| {
+                let sovereign_account =
+                    parachain_account_sovereign_account_id(1u32, account_id.clone());
+                relay_chain::Balances::make_free_balance_be(
+                    &sovereign_account,
+                    INITIAL_BALANCE,
+                );
+            });
+
+            let mut call_builder = contract.call_builder::<ContractXcm>();
+            let message = call_builder.lock_funds_to_relay(TRANSFER_VALUE, 8_000);
+            let call_res = client.call(&ink_e2e::alice(), &message).submit().await?;
+
+            assert!(call_res.return_value().is_ok());
+
+            // Check if the funds are locked on the relay chain.
+            Relay::execute_with(|| {
+                assert_eq!(
+                    relay_chain::Balances::locks(
+                        &parachain_account_sovereign_account_id(1, account_id)
+                    ),
+                    vec![BalanceLock {
+                        id: *b"py/xcmlk",
```

Comment:
Please add a comment that describes why this literal is used here and how to determine which ones to use in different scenarios.

Reply:
This comment was marked resolved, but I don't see a comment in the Files Changed. I'm unresolving hence.

Reply:
my bad, mis-clicked it, will add comments here